processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -74,6 +76,80 @@
 {
   private volatile DateTime maxIngestedEventTime;
 
+  private static final Map<String, ValueType> typeMap = ImmutableMap.<String, ValueType>builder()
+      .put("Long[]", ValueType.LONG)
+      .put("Double[]", ValueType.FLOAT)
+      .put("String[]", ValueType.STRING)
+      .put("Long", ValueType.LONG)
+      .put("Double", ValueType.FLOAT)
+      .put("String", ValueType.STRING)
+      .build();
+
+  private static final Function<Object, String>  STRING_TRANSFORMER = new Function<Object, String>()
+  {
+    @Override
+    public String apply(final Object o)
+    {
+      return String.valueOf(o);
+    }
+  };
+
+  private static final Function<Object, Long>  LONG_TRANSFORMER = new Function<Object, Long>()
+  {
+    @Override
+    public Long apply(final Object o)
+    {
+      if (o == null) {
+        return null;
+      }
+      if (o instanceof String) {
+        try {
+          return Long.valueOf((String) o);
+        } catch (NumberFormatException nfe) {
+          return null;
+        }
+      }
+      if (o instanceof Double) {
+        return ((Double) o).longValue();
+      }
+      if (o instanceof Long) {
+        return (Long) o;
+      }
```

Comment:
probably can be replaced with - 
if (metricValue instanceof Number) {
      return ((Number) metricValue).longValue();
   }